### PR TITLE
[Feature] Add mmpose in PyMAF-X

### PR DESCRIPTION
**Pull request by jiaqiAA**
Add mmpose in PyMAF-X.

### CODE REVIEW

File: demo/pymafx_estimate_smplx.py [hunk 1]
```diff
@@ -366,5 +489,16 @@ def init_openpifpaf(parser):
         type=int,
         default=8,
         help='batch size for SMPL prediction')
+    if has_openpifpaf:
+        init_openpifpaf(parser)
```
Comment (yl-1993) [inline demo/pymafx_estimate_smplx.py:493]: Moving L492-493 under L495?

File: demo/pymafx_estimate_smplx.py [hunk 2]
```diff
@@ -366,5 +489,16 @@ def init_openpifpaf(parser):
...
     args = parser.parse_args()
+    if args.use_openpifpaf:
+        assert has_openpifpaf, 'Please install openpifpaf to run the demo.'
+        assert args.det_config is not None
+        assert args.det_checkpoint is not None
+    if args.use_mmpose:
+        assert has_mmdet, 'Please install mmdet to run the demo.'
+        assert has_mmpose, 'Please install mmpose to run the demo.'
+        assert args.det_config is not None
+        assert args.pose_config is not None
```
Comment (yl-1993) [inline demo/pymafx_estimate_smplx.py:503]: Shall we `assert args.use_mmpose or args.use_openpifpaf`?

Or shall we simply remove `use_mmpose`? If users do not pass `use_openpifpaf`, we choose mmpose automatically.

File: demo/pymafx_estimate_smplx.py [hunk 3]
```diff
@@ -4,76 +4,171 @@
...
     Predictor.configure(pp_args)
     Stream.configure(pp_args)
 
     Predictor.batch_size = 1
     Predictor.loader_workers = 1
     predictor = Predictor()
-    if args.vid_file is not None:
-        capture = Stream(args.vid_file, preprocess=predictor.preprocess)
+    if Path(args.input_path).is_file():
+        image_folder = os.path.join(args.output_path, 'images')
```
Comment (yl-1993) [inline demo/pymafx_estimate_smplx.py:154]: As `osp` has been imported in L6, all `os.path` can be replaced with `osp`.

File: demo/mmpose_cfg/hrnet_w48_coco_wholebody_384x288_dark_plus.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
+load_from = 'https://download.openmmlab.com/mmpose/top_down/hrnet/hrnet_w48_coco_384x288_dark-741844ba_20200812.pth'  # noqa: E501
```
Comment (yl-1993) [inline demo/mmpose_cfg/hrnet_w48_coco_wholebody_384x288_dark_plus.py:1]: Where do we need `load_from`? It seems that `pose_checkpoint` in [demo/pymafx_estimate_smplx.py](https://github.com/open-mmlab/mmhuman3d/pull/306/files#diff-d2c32733320f01f6c05ad58cb4e2d9324ea5bc7ee76e829c5a2b171fd680f0b2R467) has already specified the checkpoint.

File: demo/pymafx_estimate_smplx.py (hunk 2 above)
Comment (jiaqiAA) [reply demo/pymafx_estimate_smplx.py:503]: I removed `use_mmpose`, and set mmpose as the default.

File: demo/mmpose_cfg/hrnet_w48_coco_wholebody_384x288_dark_plus.py (hunk 4 above)
Comment (jiaqiAA) [reply demo/mmpose_cfg/hrnet_w48_coco_wholebody_384x288_dark_plus.py:1]: I deleted it.

File: demo/mmpose_cfg/hrnet_w48_coco_wholebody_384x288_dark_plus.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
+evaluation = dict(interval=10, metric='mAP', save_best='AP')
+
+# learning policy
```
Comment (yl-1993) [inline demo/mmpose_cfg/hrnet_w48_coco_wholebody_384x288_dark_plus.py:3]: `learning policy` is also not necessary for inference?

**Review by yl-1993 — APPROVED** [review]

**Pull request merged**